validator/db/kv/attestation_history_v2.go

```diff
@@ -46,9 +47,10 @@ func (store *Store) AttestationHistoryForPubKeyV2(ctx context.Context, publicKey
 		enc := bucket.Get(publicKey[:])
 		if len(enc) == 0 {
 			attestationHistory = NewAttestationHistoryArray(0)
+		} else if data, err := snappy.Decode(nil /*dst*/, enc); err != nil {
+			return err
 		} else {
-			attestationHistory = make(EncHistoryData, len(enc))
-			copy(attestationHistory, enc)
+			attestationHistory = data
 		}
```

Comment:
```go
                else {
			data, err := snappy.Decode(nil /*dst*/, enc)
			if err != nil {
				return err
			}
			attestationHistory = data
		}
```

Is this more clear? You can lose the `else if`

Reply:
I see how this is more clear. I'll fix, thanks!